### PR TITLE
Configurable for agent svid ttl

### PR DESCRIPTION
**Pull request by evan2645**
SPIRE Server currently has the `default_svid_ttl` configurable, which
dictates the TTL that should be used when signing SVIDs where a TTL is
not otherwise specified.

One quirk of this configurable is that since the SVIDs that agents use
are not minted against a registration entry, the default SVID TTL is
used. Since an agent's SVID TTL also constrains how long it can be
offline or unavailable for, it is a fairly important number that most
users put thought into and directly configure.

Since it can be desirable to have agents that can survive long(er)
periods offline, while still having a low default TTL for workload
SVIDs, it makes sense to allow agent SVID TTLs to be separately
configurable.

This commit introduces an `agent_ttl` configurable on SPIRE Server. If
configured, all agent SVIDs will be minted with this TTL. If not
configured, SPIRE will fall back to the default SVID TTL.

Signed-off-by: Evan Gilman <egilman@vmware.com>

### CODE REVIEW

File: pkg/server/api/agent/v1/service.go [hunk 1]
```diff
@@ -50,6 +51,8 @@ type Service struct {
 	ds  datastore.DataStore
 	ca  ca.ServerCA
 	td  spiffeid.TrustDomain
+
```
Comment (azdagron) [inline pkg/server/api/agent/v1/service.go:54]: nit: can we group this with the rest of the fields? any reason to keep it separate?

File: pkg/server/api/agent/v1/service.go [hunk 2]
```diff
@@ -60,6 +63,8 @@ func New(config Config) *Service {
 		ds:  config.DataStore,
 		ca:  config.ServerCA,
 		td:  config.TrustDomain,
+
```
Comment (azdagron) [inline pkg/server/api/agent/v1/service.go:66]: nit: same here

File: pkg/server/api/agent/v1/service_test.go [hunk 3]
```diff
@@ -2989,6 +2996,8 @@ func setupServiceTest(t *testing.T) *serviceTest {
 		TrustDomain: td,
 		Clock:       clk,
 		Catalog:     cat,
+
+		AgentTTL: agentTTL,
```
Comment (azdagron) [inline pkg/server/api/agent/v1/service_test.go:3000]: nit: same here

**Review by azdagron — APPROVED** [review]
Looks great! Nothing but nits....

**Comment by azdagron** [comment]
Oh, except we should probably document this in server_full.conf....

File: conf/server/server_full.conf [hunk 4]
```diff
@@ -117,8 +117,10 @@ server {
     # Default: /tmp/spire-server/private/api.sock.
     # socket_path = "/tmp/spire-server/private/api.sock"
 
-    # default_svid_ttl: The default SVID TTL. Default: 1h.
```
Comment (azdagron) [inline conf/server/server_full.conf:120]: Don't we need to keep this?

Comment (evan2645) [reply conf/server/server_full.conf:120]: lol yep already fixed

**Review by azdagron — APPROVED** [review]
\o/, thanks @evan2645 !

**Pull request merged**